cpp/src/arrow/compute/api_scalar.h

```diff
@@ -316,6 +316,21 @@ struct ARROW_EXPORT CompareOptions {
   enum CompareOperator op;
 };
 
+enum class BetweenMode : int8_t {
+  LESS_EQUAL_LESS_EQUAL,
+  LESS_EQUAL_LESS,
+  LESS_LESS_EQUAL,
+  LESS_LESS,
+};
```

Comment:
Done.

Reply:
Hmm, @edponce is this structure what you expected? I think for consistency with other options it would look like this:

```cpp
class BetweenOptions {
 public:
  enum Inclusive {
    BOTH,
    // ...
  };
  // ...
};
```

...where `enum` is deliberately used over `enum class` so that the enum members can be accessed as `BetweenOptions::BOTH`.

Reply:
Yes, I agree with @lidavidm on how the BetweenOptions should be organized.

Reply:
@lidavidm Thanks for the feedback. Inclusive is in global space so that it can also be used in NotBetween - follow up issue https://issues.apache.org/jira/browse/ARROW-15223 , unless NotBetween should be implemented as an option of Between.

Reply:
not_between can also use BetweenOptions, it doesn't need its own options class unless I'm mistaken.

Reply:
@lidavidm is correct. There should only be one:
```c++
class BetweenOptions {
 public:
  enum Inclusive {
    BOTH,
    // ...
  };
  // ...
};
```
in global space, so that it can be used  by both `between` and `not_between`. Something important to note is that[ there is a discussion in JIRA on whether "not_between" should exist](https://issues.apache.org/jira/browse/ARROW-15223).